diff --git a/mcp_can.cpp b/mcp_can.cpp
index 5507db0..49830d9 100644
--- a/mcp_can.cpp
+++ b/mcp_can.cpp
@@ -50,7 +50,7 @@
 #define spi_readwrite      pSPI->transfer
 #define spi_read()         spi_readwrite(0x00)
 #define spi_write(spi_val) spi_readwrite(spi_val)
-#define SPI_BEGIN()        pSPI->beginTransaction(SPISettings(10000000, MSBFIRST, SPI_MODE0))
+#define SPI_BEGIN()        pSPI->beginTransaction(SPISettings(2000000, MSBFIRST, SPI_MODE0))
 #define SPI_END()          pSPI->endTransaction()
 
 /*********************************************************************************************************
